NR46306: Lighttpd-new integration

dikshab0202 opened the pull request:
Ticket: [NR-46306](https://issues.newrelic.com/browse/NR-46306)
Description: Lighttpd - New integration

Preview image:
![image](https://user-images.githubusercontent.com/101187392/220082575-9c9fbeb7-c89f-4d03-b9be-3022d7503081.png)


# Summary

A concise description of the changes being introduced. Please review the pre-merge checklist section to validate this pull request is ready for review and merge. **If it is not ready, please mark the pull request as a draft.**

<!-- DON'T DELETE THIS SECTION BELOW IF SUBMITTING A NEW QUICKSTART -->
## Pre merge checklist

<!-- THIS CHECKLIST MUST BE FULLY COMPLETE OR YOUR PR WILL NOT BE MERGED -->

- [ ] Did you check you NRQL syntax? - Does it work?
- [ ] Did you include an InstallPlan and Documentation reference?
- [ ] Are all documentation links publically accessible?
- [ ] Did you check your descriptive content for [voice and tone](https://docs.newrelic.com/docs/style-guide/writing-strategies/voice-strategies-docs-sound-new-relic/)? 
- [ ] Did you check your descriptive content for spelling and grammar errors?

### Dashboards 

- [ ] Does the PR contain a screenshot for each of your dashboards?
- [ ] Do your screenshots show data?
- [ ] Has the [sanitization script](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#dashboards) been run on each dashboard?

### Alerts

- [ ] Did you check that your alerts actually work?

dikshab0202 commented:
Waiting for Lighttpd doc url.

paperclypse commented on alert-policies/lighttpd/AverageNumberOfConnections.yml:
```diff
@@ -0,0 +1,36 @@
+name: Average Number of Connections
+
+description: |+
+  This alert is triggered when the average number of connections are above 6000 for atleast 2 minutes.
```
```suggestion
  This alert is triggered when the average number of connections are above 6000 for at least 2 minutes.
```

paperclypse commented on alert-policies/lighttpd/AverageNumberOfConnections.yml:
```diff
@@ -0,0 +1,36 @@
+    thresholdOccurrences: ALL
+
+  - priority: WARNING
+    # Operator used to compare against the threshold.
```
Why do some of these comments end in periods and not others. I prefer consistency :)

paperclypse commented on alert-policies/lighttpd/AvgNumberOfRequestsFor5sec.yml:
```diff
@@ -0,0 +1,36 @@
+name: Average Number of Requests for 5 Seconds
+
+description: |+
+  This alert is triggered when the average number of requests for 5 seconds are above 6000 for atleast 5 minutes.
```
```suggestion
  This alert is triggered when the average number of requests for 5 seconds are above 6000 for at least 5 minutes.
```

paperclypse commented on alert-policies/lighttpd/HighMemoryUsage.yml:
```diff
@@ -0,0 +1,36 @@
+name: High Memory Usage
+
+description: |+
+  This alert is triggered when the memory usage is above 90% for atleast 5 minutes.
```
```suggestion
  This alert is triggered when the memory usage is above 90% for at least 5 minutes.
```

paperclypse commented on install/lighttpd/install.yml:
```diff
@@ -0,0 +1,15 @@
+name: lighttpd
+title: Lighttpd
+description: |
+  To collect data and deliver it to New Relic, integrate your Lighttpd processes with our Infrastructure agent and Flex configuration.
```
```suggestion
  To collect data and deliver it to New Relic, integrate your lighttpd processes with our Infrastructure agent and Flex configuration.
```

Based on its website, lighttpd is always lowecase

paperclypse replied:
Also, did you know lighttpd is pronounced _lighty_? :)

dikshab0202 replied:
Yes, Shawn.

paperclypse replied:
I didn't :) I was very surprised!

paperclypse commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,40 @@
+description: |
+  ## Comprehensive monitoring quickstart for Lighttpd
+
+  Lighttpd is an open-source web server designed for high-performance situations while staying standards-compliant, secure and adaptable.
```
```suggestion
  Lighttpd is an open-source web server designed for high-performance situations while staying standards-compliant, secure, and adaptable.
```

This is a great intro sentence! We prefer the Oxford (or serial) comma though :)

paperclypse commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,40 @@
+
+  Lighttpd is an open-source web server designed for high-performance situations while staying standards-compliant, secure and adaptable.
+
+  ### Why monitor Lighttpd?
```
```suggestion
  ### Why monitor your lighttpd server?
```

paperclypse commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,40 @@
+
+  ### Why monitor Lighttpd?
+
+  Lighttpd monitoring provides important information to help the developers construct a complete picture of a web server's performance. This data comprises uptime, network in bytes and packets, number of connections and more.
```
```suggestion
  Lighttpd monitoring provides important information to help you construct a complete picture of your web server's performance. The performance data includes uptime, network in bytes and packets, number of connections, and more.
```

paperclypse commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,40 @@
+
+  ### What’s included?
+
+  The Lighttpd monitoring quickstart from New Relic provides immediate full-stack observability:
```
```suggestion
  When you monitor your lighttpd server withNew Relic, you get immediate full-stack observability, including:
```

paperclypse commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,40 @@
+  The Lighttpd monitoring quickstart from New Relic provides immediate full-stack observability:
+  - Alerts (average number of connections, average number of requests, high memory usage).
+  - Dashboards (uptime, connection state handle request, connection state still alive, networks, requests per second and more).
+  - Be proactive in monitoring potential issues and resolving them as soon as possible.
```
I'd delete this last bullet. It doesn't seem very helpful or informative.

If we must keep it, I'd rewrite it to:

"Deeper insight into your server's performance issues."

paperclypse commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,40 @@
+  - Be proactive in monitoring potential issues and resolving them as soon as possible.
+
+summary: |
+   With our Infrastructure agent and Flex configuration setup, you can monitor and validate your Lighttpd processes. Right out of the box, analyze and capture performance data.
```
```suggestion
   With our infrastructure agent and Flex configuration setup, you can monitor and validate your lighttpd processes. Right out of the box, analyze and capture your server's performance data.
```

dikshab0202 replied:
Flex is an application-agnostic, all-in-one [New Relic integration](https://docs.newrelic.com/docs/integrations) with which you can instrument any app that exposes metrics over a standard protocol (HTTP, file, shell) in a standard format.
 Sharing [github url](https://github.com/newrelic/nri-flex) for the better understanding.

paperclypse replied:
Thank you!

paperclypse reviewed (dismissed):
I left some comments and suggested edits.

paperclypse commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,39 @@
+description: |
+  ## Comprehensive monitoring for your lighttpd server
+
+  lighttpd is an open-source web server designed for high-performance situations while staying standards-compliant, secure, and adaptable.
```
lighttpd should still be capitalized when it's the start of a sentence.

dikshab0202 replied:
okay @paperclypse . I applied the above changes. 
If possible can you please once check [lighttpd page](https://www.lighttpd.net/), I found there it started with small letter in the starting of a sentence as well.

paperclypse replied:
For the start of a sentence, I'd prefer to override their style choice (lowercase at start of sentence) with ours (always capitalizing the start of a sentence). I looked through their docs and they're not consistent on this point, so I feel pretty okay with capitalizing it at the start of a sentence.

dikshab0202 replied:
Okay @paperclypse, no problem. I have applied all the changes and pushed. Kindly review and let me know if any more suggestions or changes required.
Thank you.

paperclypse replied:
Looks good to me, thanks!

paperclypse reviewed (approved):
Looks good!

brammerl commented:
Hello, We are no longer accepting install plans as a part of the quickstart schema. We are instead moving to use data sources. You can either refer to an [already existing data source](https://github.com/newrelic/newrelic-quickstarts/tree/main/data-sources) in this repository or[ make your own](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#data-sources-1) community data source.

If this applies to your PR, please make the proper update to your quickstart's config to reference only data sources.

dikshab0202 commented:
> Hello, We are no longer accepting install plans as a part of the quickstart schema. We are instead moving to use data sources. You can either refer to an [already existing data source](https://github.com/newrelic/newrelic-quickstarts/tree/main/data-sources) in this repository or[ make your own](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#data-sources-1) community data source.
> 
> If this applies to your PR, please make the proper update to your quickstart's config to reference only data sources.

Thank you @brammerl . I have updated this PR by adding datasources files and removed install plan. Could you please review and let me know if any changes are required.

pkosanam reviewed (approved):
looks good

pkosanam commented on data-sources/lighttpd/config.yml:
```diff
@@ -0,0 +1,22 @@
+id: lighttpd
```
Looks good..
There are few more fields which are to be added.

Request you to kindly follow this contribution guide
guide: https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#data-sources-1
Thank you 😃

pkosanam reviewed (dismissed):
looks good

seemantk commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,41 @@
+  ### What’s included?
+
+  When you monitor your lighttpd server with New Relic, you get immediate full-stack observability, including:
+  - Be alerted on things like average number of connections, average number of requests and high memory usage.
```
Be alerted on things like -> Alerts on

dikshab0202 replied:
@seemantk , Changes have been completed. Kindly review and let me know if something incorrect.
Thank you..

brammerl commented on data-sources/lighttpd/config.yml:
```diff
@@ -0,0 +1,22 @@
+id: lighttpd
+displayName: Lighttpd
+description: |
+  ## Comprehensive monitoring for your lighttpd server
```
Data source descriptions don't support markdown and are expected to be a short description. Please come up with a short/concise description for this data source

seemantk commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,41 @@
+
+  When you monitor your lighttpd server with New Relic, you get immediate full-stack observability, including:
+  - Be alerted on things like average number of connections, average number of requests and high memory usage.
+  - Use dashboards to monitor crucial performance metrics, such as uptime, connection state handle request, connection state still alive, networks, requests per second, and more.
```
Use dashboards -> Dashboards
performance metrics, -> performance metrics

brammerl commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,41 @@
+
```
Let's remove this space

seemantk commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,41 @@
+  - Use dashboards to monitor crucial performance metrics, such as uptime, connection state handle request, connection state still alive, networks, requests per second, and more.
+
+summary: |
+   With our infrastructure agent and Flex configuration setup, you can monitor and validate your lighttpd processes. Right out of the box, analyze and capture your server's performance data.
```
"lighttpd processes. Right out of the box," -> "lighttpd processes right out of the box:"

dikshab0202 commented on quickstarts/lighttpd/config.yml:
```diff
@@ -0,0 +1,41 @@
+  - guided-install
+dataSourceIds:
+  - lighttpd
+keywords:
```
Hi @brammerl , Thank you for the information. I did these changes. Kindly review and let me know if something incorrect.

pkosanam reviewed (approved):
looks good
